### DISCUSSION TITLE
delete sandbox

### DISCUSSION THREAD

**Discussion by jammywwh** [sylabs/singularity#2938] [Q&A]
Hi,
As a user, the administrator has granted me the build permission of singularity.
After I used the following command to build the sandbox,
I wanted to delete the folder where this sandbox is located, but I found that this folder belongs to root and I don't have the permission to delete it. What should I do?

```
sudo singularity build --sandbox ./tempDir annotsv.def
sudo rm -rf ./tempDir
```

It is expected that this folder (./tempDir) can be deleted.
Actual behavior: Permission denied

Version of Singularity: 3.8.7-1.el7
OS/distro:
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Comment by dtrudg** [comment]
Hello. This issue has been converted to a discussion as it is a question around usage, rather than a bug in SIngularity.

I'm not clear exactly what permission you have been granted? Has the administrator only allowed you to run `singularity build` with `sudo` - or have you been permitted to run all commands?

If you are not allowed to run `rm` with `sudo` then its expected you cannot remove the directory. By running the build with `sudo` it will always be owned by root.

You may be able to partly get around this by:

1) Running the build with sudo, to create a .sif file
2) Converting the .sif file into a sandbox as your own user, without sudo.

```
sudo singularity build temp.sif annotsv.def
singularity build -s ./tempDir annotsv.def
```

The .sif file would still be owned by root, but the sandbox would then be owned by your user account.

For the best advice it's better to give full information about what you are trying to acheive. What is the workflow that you are trying to follow? Why do you need a sandbox?

**Reply by ExplorerRay** [reply]
Hi, I also have some problems related to this. 

In my scenario, I am a user who doesn't have sudo, so I run `singularity build --sandbox test/ library://rockylinux` and try to manually build my image under my home directory without using fakeroot.

However, after doing so, when I want to delete `test/` directory with `rm -rf test/`, it says `Permission denied` even if I am the owner of those files (particularly `test/root` and `test/usr`)

update: I also try ubuntu image and it delete successfully.

**Reply by dtrudg** [reply]
When using `--fakeroot`, files may be created with subuid / subgid mappings configured for your account. These IDs are only available to you when inside a user namespace.

 You can use `rm` inside of a fakeroot shell to remove the files, e.g.

```
$ singularity shell --fakeroot docker://alpine
Singularity> rm -rf test/
```

Alternative, if the issue is that the files are visibly owned by your account on the host, but do not have owner rwX permissions, you can `chmod` them before deleting, or use the `--fixperms` flag when performing the `singularity build`.
